resources/device_probe.py

```diff
@@ -880,6 +884,33 @@ def bluetooth_probe(self):
         elif any("Bluetooth" in usb_device.product_name for usb_device in self.usb_devices):
             self.bluetooth_chipset = "Generic"
 
+    def keyboard_probe(self):
+        if not self.usb_devices:
+            return
+
+        legacy_ids = [526,527,528,532,533,534,525,536,537,538,539,540,553,554,555] # These keyboards were stripped of their personalities after being dropped
+        modern_ids = [547,548,549,560,561,562,566,567,568,575,576,577,578,579,580,581,582,583,585,586,587,601,602,603,610,611,612,588,589,590,594,595,596] #These keyboards remained in AppleUSBTCKeyboard until the kext was removed
```

Comment:
I'm wondering if we should create a new data library to store these, similar to `data/pci_data.py`. Would allow us to reference them elsewhere, as well as have a centralized area to reference them if we start adding more IDs (ex. Bluetooth)

Something like `data/usb_data.py`:
```py
class AppleIDs:

   Modern_AppleUSBTCKeyboard = [
      526,
      # etc...
   ]

   Legacy_AppleUSBTCKeyboard = [
      # etc...
   ]
```

Reply:
+1